### PR TITLE
update with support for torch.bmm in flop_count

### LINKED ISSUE #49: flop_count support more ops, such as bmm
Now `flop_count` already supports several ops with high computational costs:
https://github.com/facebookresearch/fvcore/blob/1dbd159daebf39dd1cb5911d5ab9b64418d0057e/fvcore/nn/flop_count.py#L20-L25
However with the popularity of `MultiHeadAttention`, some ops with high computational costs are not taken into consideration, such as `bmm`. Some element-wise ops can also be costly such as `mul`, `add` and `layer_norm`.
image
Will `flop_count` support these ops in the future?

### PR DESCRIPTION
**Pull request by tonysy**
close #49

### CODE REVIEW

File: fvcore/nn/jit_handles.py [hunk 1]
```diff
@@ -202,6 +202,34 @@ def addmm_flop_jit(
     return flop_counter
 
 
+def bmm_flop_jit(
+    inputs: typing.List[object], outputs: typing.List[object]
+) -> typing.Counter[str]:
+    """
+    This method counts the flops for the bmm operation.
+    We currently support the operation: "nct,ntd->ntd" .
```
Comment (ppwwyyxx) [inline fvcore/nn/jit_handles.py:210]: This is redundant becaue it is what bmm means
```suggestion
```

File: fvcore/nn/jit_handles.py [hunk 2]
```diff
@@ -202,6 +202,34 @@ def addmm_flop_jit(
...
+
+    Args:
+        inputs (list(torch._C.Value)): The input shape in the form of a list of
+            jit object before einsum.
+        outputs (list(torch._C.Value)): The output shape in the form of a list
+            of jit object after einsum.
```
Comment (ppwwyyxx) [inline fvcore/nn/jit_handles.py:216]: einsum -> bmm

File: tests/test_flop_count.py [hunk 3]
```diff
@@ -112,6 +112,20 @@ def forward(self, x: torch.Tensor, y: torch.Tensor) -> torch.Tensor:
...
 
+class BMMNet(nn.Module):
+    """
+    A network with a single torch.bmm operation. This is used for testing
+    flop count for torch.bmm.
+    """
+
+    def __init__(self) -> None:
+        super(BMMNet, self).__init__()
+
```
Comment (ppwwyyxx) [inline tests/test_flop_count.py:123]:
```suggestion
```

Comment (tonysy) [reply tests/test_flop_count.py:123]: Hi, remove the blank line(line-123) will raise the format check error.

Comment (ppwwyyxx) [reply tests/test_flop_count.py:123]: Oh I mean the init method can be removed

Comment (tonysy) [reply tests/test_flop_count.py]: Done. (By the way, the init method of MatmulNet can be removed also)

**Comment by tonysy** [comment]
Hi, what I can do to handle the failure of the internal test in checks

**Comment by ppwwyyxx** [comment]
No need to worry about that!

**Pull request closed without merging**